File: c/Space-age.c
//space_age.c
#include "space_age.h"
float orbital[8]={0.2408467,0.61519726,1.0,1.8808158,11.862615,29.447498,84.016846,164.79132};
float age(planet_t planet, int64_t seconds)
{  
    if ((planet>=0) && (planet<8))
    {
        float earth_year = seconds / 31557600;
        return earth_year/orbital[planet];
    }
    else{return -1;}
   
}
/*
#ifndef SPACE_AGE_H
#define SPACE_AGE_H

#include <stdint.h>

typedef enum planet {
   MERCURY,
   VENUS,
   EARTH,
   MARS,
   JUPITER,
   SATURN,
   URANUS,
   NEPTUNE,
} planet_t;

float age(planet_t planet, int64_t seconds);

#endif

*/

File: javascript/SpaceAge.js
const RATIO = {
    mercury: 0.2408467,
    venus: 0.61519726,
    earth: 1,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132
},
EARTH_YEAR = 31557600

export function age(planet, seconds) {
    // Koşulu burada kontrol et: planet, RATIO objesinde mevcut mu?
    if (!RATIO[planet]) { // Eğer RATIO objesinde böyle bir gezegen yoksa (tanımsızsa)
        throw new Error("not a planet"); // <-- Bu hatayı fırlat!
    }

    // Eğer gezegen varsa, hesaplamaya devam et
    return Number((seconds / EARTH_YEAR / RATIO[planet]).toFixed(2));
}